docs/src/snap/howto/storage/cloud.md

```diff
@@ -0,0 +1,486 @@
+        "elasticloadbalancing:DescribeLoadBalancerPolicies",
+        "elasticloadbalancing:DescribeTargetGroups",
+        "elasticloadbalancing:DescribeTargetHealth",
+        "elasticloadbalancing:ModifyListener",
+        "elasticloadbalancing:ModifyTargetGroup",
+        "elasticloadbalancing:RegisterTargets",
+        "elasticloadbalancing:DeregisterTargets",
+        "elasticloadbalancing:SetLoadBalancerPoliciesOfListener",
+        "iam:CreateServiceLinkedRole",
+        "kms:DescribeKey"
+      ],
+      "Resource": [
+        "*"
+      ]
+    }
+  ]
+}
+```
+
+For a worker node:
+
+```{dropdown} Worker Policies
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeInstances",
+        "ec2:DescribeRegions",
+        "ecr:GetAuthorizationToken",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:GetRepositoryPolicy",
+        "ecr:DescribeRepositories",
+        "ecr:ListImages",
+        "ecr:BatchGetImage"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+## Add a tag to your EC2 Instance
+
+A cluster using the AWS cloud provider needs to label existing nodes and
+resources with a ClusterID or the kube-controller-manager will not start. Add
+the following tag to your instance, making sure to replace the placeholder id
+with your own (this can simply be "k8s" or "my-k8s-cluster").
+
+```
+kubernetes.io/cluster/<your-cluster-id>=owned
+```
+
+## Set your host name
+
+The cloud controller manager uses the node name to correctly associate the node
+with an EC2 instance. In {{product}}, the node name is derived from the
+hostname of the machine. Therefore, before bootstrapping the cluster, we must
+first set an appropriate host name.
+
+```bash
+echo "$(sudo cloud-init query ds.meta_data.local-hostname)" | sudo tee /etc/hostname
+```
+
+Then, reboot the machine.
+
+```bash
+sudo reboot
+```
+
+When the machine is up, use `hostname -f` to check the host name. It should
+look like:
+
+```bash
+ip-172-31-11-86.us-east-2.compute.internal
+```
+
+This host name format is called IP-based naming and is specific to AWS.
+
+
+## Bootstrap {{product}}
+
+Now that your machine has an appropriate host name, you are ready to bootstrap
+{{product}}.
+
+First, create a bootstrap configuration file that sets the cloud-provider
+configuration to "external".
+
+```bash
+echo "cluster-config:
+  cloud-provider: external" > bootstrap-config.yaml
+```
+
+Then, bootstrap the cluster:
+
+```bash
+sudo k8s bootstrap --file ./bootstrap-config.yaml
+sudo k8s status --wait-ready
+```
+
+## Deploy the cloud controller manager
+
+Now that you have an appropriate host name, policies, and a {{product}}
+cluster, you have everything you need to deploy the cloud controller manager.
+
+Here is a YAML definition file that sets appropriate defaults for you, it
+configures the necessary service accounts, roles, and daemonsets:
+
```

Comment:
should you include a kubectl apply -f command to deploy the controller manager?

Reply:
Ah, that was an oversight on my part. Thanks.